docs/src/lib/types.md

```diff
@@ -9,11 +9,33 @@ CurrentModule = DataFrames
 Pages = ["types.md"]
 ```
 
+## Type hierarchy design
+
+`AbstractDataFrame` is an abstract type that provides database-like interface.
```

Comment:
...interface for representing data in relational format.

Database-like seems too broad.